### PR TITLE
fix: Restructure dependencies and fix for launchbar not visible issues

### PR DESCRIPTION
**Pull request by kolipakakondal**
## Description

Restructure dependencies and fix for launchbar not visible issues

Fixes # ([IEP-XXX](https://jira.espressif.com:8443/browse/IEP-XXX))

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How has this been tested?

- Able to install PR update site on eclipse and espressif-ide v2.12.1 
- Able to install PR update site on espressif-ide v2.12.1 and launchbar is visible

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Update site
- Epressif-ide installation

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
	- Enhanced support for external libraries and dependencies, ensuring better performance and compatibility.
- **Refactor**
	- Updated product configuration to support a mix of features for improved flexibility.
	- Revised dependency management for a more streamlined and efficient development environment.
- **Chores**
	- Updated category names for bundles to better reflect their purpose and content.
	- Improved categorization by replacing "com.espressif.idf" with "com.espressif.idf.dependencies" for multiple bundles.

### CODE REVIEW

File: releng/com.espressif.idf.product/idf.product [hunk 1]
```diff
@@ -76,6 +76,7 @@
       <feature id="org.eclipse.swtchart.feature" installMode="root"/>
       <feature id="org.eclipse.nebula.widgets.xviewer.feature" installMode="root"/>
       <feature id="org.eclipse.cdt.lsp.feature" installMode="root"/>
+      <feature id="org.eclipse.tm4e.feature" installMode="root"/>
```
[the remark this answers is not in the document]
Comment (kolipakakondal) [reply releng/com.espressif.idf.product/idf.product:79]: Yes

**Comment by AndriiFilippov** [comment]
@kolipakakondal  @sigmaaa  hi ! 

Tested under: 

OS - Windows 10 
ESP-IDF: v5.2 

Do not see toolbar. 
after installation of org.apache.batik.* plugins - do see toolbar. 
image


image

**Comment by kolipakakondal** [comment]
Hi @AndriiFilippov It's fixed, could you check now.

**Comment by AndriiFilippov** [comment]
@kolipakakondal hi !

Do see toolbar 
image

still visible after restart or exit IDE. 

LGTM 👍

**Pull request merged**